core/schedule.go

```diff
@@ -27,9 +27,15 @@ import (
 	"github.com/intelsdi-x/snap/pkg/schedule"
 )
 
+// Schedule defines a scheduler.
+//
+// swagger:model Schedule
 type Schedule struct {
-	Type           string     `json:"type,omitempty"`
-	Interval       string     `json:"interval,omitempty"`
+	// required: true
+	// enum: simple, windowed, streaming, cron
+	Type string `json:"type"`
```

Comment:
it's beneficial that users can select a type from an available GUI list instead of by leaving it empty although we could default it to `simple`. It's not clear when we have public exposing APIs.